python/ml4ir/applications/classification/model/metrics/metrics_impl.py

```diff
@@ -107,5 +107,5 @@ def update_state(self, y_true, y_pred, sample_weight=None):
         will throw error.
         """
         return super(Top5CategoricalAccuracy, self).update_state(
-            tf.squeeze(y_true, axis=1), tf.squeeze(y_pred, axis=1), sample_weight=sample_weight
+            tf.squeeze(y_true), tf.squeeze(y_pred), sample_weight=sample_weight
```

Comment:
tf.sqeeze will only remove dimensions of 1. this makes the code here robust, keeping the standard behaviour of the metrics (according to TF) as well as making it compatible with the extra dimension we have.